modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/snapshot/MajorityPeerSelect.scala

```diff
@@ -50,28 +60,32 @@ object MajorityPeerSelect {
 
     def filterPeerList(peers: NonEmptyList[Peer]): F[NonEmptyList[Peer]] =
       peers
-        .traverse(snapshotClient.getLatestOrdinal(_))
-        .map {
-          _.groupBy(identity).maxBy { case (_, ordinals) => ordinals.size }
+        .parTraverseN(maxConcurrentPeerInquiries)(snapshotClient.getLatestOrdinal(_))
+        .map(_.groupBy(identity).maxBy { case (_, ordinals) => ordinals.size })
+        .map { case (majorityOrdinal, _) => majorityOrdinal }
+        .flatMap { majorityOrdinal =>
+          peers
+            .parTraverseN(maxConcurrentPeerInquiries)(peer => getPeerHash(peer, majorityOrdinal).attempt)
         }
-        .flatMap {
-          case (majorityOrdinal, _) =>
-            peers.traverse(snapshotClient.get(majorityOrdinal).run(_).flatMap(_.toHashed.map(_.hash)))
+        .flatMap { maybePeerSnapshots =>
+          MonadThrow[F].fromOption(
+            maybePeerSnapshots.collect { case Right(peerSnapshot) => peerSnapshot }.toNel,
+            NoPeersToSelect
+          )
         }
-        .map(_.zip(peers))
-        .map(_.groupMap { case (hash, _) => hash } { case (_, ps) => ps })
-        .map(_.maxBy { case (_, peers) => peers.size })
-        .map { case (_, peerCandidates) => peerCandidates }
+        .map(_.groupMap { case (_, hash) => hash } { case (peer, _) => peer })
+        .map(_.values.toList.sortBy(-_.size))
+        .map(_.head)
 
     def getPeerSublist(peers: Set[Peer]): F[List[Peer]] = {
-      val maxSublistPercent = 0.25
-      val maxSublistSize = Math.max((peers.size * maxSublistPercent).toInt, 1)
+      val sampleSize = (peers.size * peerSampleRatio).toInt
 
-      Random[F].nextIntBounded(maxSublistSize).flatMap { peerCount =>
-        Random[F]
-          .shuffleList(peers.toList)
-          .map(_.take(peerCount))
-      }
+      Random[F]
+        .shuffleList(peers.toList)
+        .map(_.take(Math.max(sampleSize, minSampleSize)))
     }
+
+    def getPeerHash(peer: Peer, ordinal: SnapshotOrdinal): F[(Peer, Hash)] =
```

Comment:
the function name suggests that it returns a peer hash but in fact it returns a snapshot hash from this peer. you cannot guess that from the type as well. IMO the name is confusing - could you rename it? `getSnapshotHash` or `getSnapshotHashByPeer` or `getSnapshotHashFromPeer`, or anything else

Reply:
Renamed to `getSnapshotHashByPeer`.